### PR TITLE
Refactor IAP and reused for frontend

### PR DESCRIPTION
**Pull request by blackzlq**
1. Make Helm general.
2. Onboard the frontend to IAP
3. Fix the permission issue for jupyter hub

Tested-by: zlq

Tested:
Sandbox

### CODE REVIEW

File: modules/jupyter/outputs.tf [hunk 1]
```diff
@@ -12,12 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
-output "jupyterhub_uri" {
```
Comment (imreddy13) [inline modules/jupyter/outputs.tf:16]: pretty sure this is wrong. did you test it works with and without IAP?

File: modules/iap/charts/iap/templates/iap-secret.yaml
Comment (imreddy13) [inline modules/iap/charts/iap/templates/iap-secret.yaml:1]: why does this show up as an added file instead of moved?

File: modules/iap/charts/iap/values.yaml [hunk 2]
```diff
@@ -6,20 +6,20 @@
...
   managedCertificate:
     name: "iap-managed-cert"
     domain: ""
 
-  jupyterIngress:
-    name: "jupyter-ingress"
+  ingress:
+    name: "iap-ingress"
     staticIpName: "xyz"
-    managedCertificates: "iap-managed-cert"
```
Comment (imreddy13) [inline modules/iap/charts/iap/values.yaml:22]: why did we delete the cert?

File: modules/iap/iap.tf
Comment (imreddy13) [inline modules/iap/iap.tf:1]: same, why is this added instead of moved? makes it harder to review.

File: modules/jupyter/outputs.tf [hunk 3]
```diff
@@ -12,12 +12,11 @@
...
-  value = var.add_auth ? module.iap_auth[0].domain : (data.kubernetes_service.jupyter-ingress.status != null ? (data.kubernetes_service.jupyter-ingress.status[0].load_balancer != null ? "${data.kubernetes_service.jupyter-ingress.status[0].load_balancer[0].ingress[0].ip}" : "") : "")
+output "jupyter_domain" {
+  value = var.add_auth ? module.iap_auth[0].jupyter_domain : var.url_domain_addr
 }
 
-output "jupyterhub_password" {
+output "jupyter_password" {
```
Comment (umeshkumhar) [inline modules/jupyter/outputs.tf:19]: can we keep the same name for outputs.. they are refered at multiple other places as well

File: applications/rag/frontend/main.tf [hunk 4]
```diff
@@ -30,6 +34,46 @@ locals {
...
+resource "google_project_service" "project_service" {
+  project = var.project_id
+  service = "iap.googleapis.com"
+
+  disable_dependent_services = false
+  disable_on_destroy         = false
+}
+
+# IAP Section: Creates the OAuth client used in IAP
+resource "google_iap_client" "iap_oauth_client" {
```
Comment (imreddy13) [inline applications/rag/frontend/main.tf:47]: @blackzlq @bjornsen do we need separate clients for frontend/jupyter? can they share one to make the setup simple?

File: applications/jupyter/outputs.tf [hunk 5]
```diff
@@ -13,10 +13,10 @@
 # limitations under the License.
 
 output "jupyterhub_uri" {
-  value = "http://${module.jupyterhub.jupyterhub_uri}"
+  value = "http://${module.jupyterhub.jupyter_domain}"
```
Comment (umeshkumhar) [inline applications/jupyter/outputs.tf:16]: same here

File: applications/rag/frontend/main.tf [hunk 6]
```diff
@@ -30,6 +34,46 @@ locals {
     instance_connection_name = format("%s:%s:%s", var.project_id, var.region, "pgvector-instance")
 }
 
+# IAP Section: Enabled the IAP service
+resource "google_project_service" "project_service" {
```
Comment (imreddy13) [inline applications/rag/frontend/main.tf:38]: can this be done in the common iap.tf module?

File: applications/rag/frontend/main.tf [hunk 7]
```diff
@@ -48,13 +92,13 @@ resource "kubernetes_service" "rag_frontend_service" {
...
     }
-    session_affinity = "ClientIP"
+    session_affinity = "None"
     port {
       port        = 8080
       target_port = 8080
     }
 
-    type = "ClusterIP"
+    type = "NodePort"
```
Comment (imreddy13) [inline applications/rag/frontend/main.tf:101]: can we have a flag to allow ClusterIP + port forward?

File: modules/iap/charts/iap/values.yaml (hunk 2 above)
Comment (blackzlq) [reply modules/iap/charts/iap/values.yaml:22]: It is duplicated with  managedCertificate. name: "iap-managed-cert". They should auto linked together.

File: modules/iap/charts/iap/templates/iap-secret.yaml
Comment (blackzlq) [reply modules/iap/charts/iap/templates/iap-secret.yaml:1]: I think because this file missing the license, I added the license so the system consider it too many diff

File: applications/rag/frontend/main.tf (hunk 7 above)
Comment (blackzlq) [reply applications/rag/frontend/main.tf:101]: Will do, discussed offline, we will highly recommend the IAP

File: applications/rag/frontend/main.tf (hunk 4 above)
Comment (blackzlq) [reply applications/rag/frontend/main.tf:47]: Discussed offline, the user case for frontend and jupyter may different, so we should do that.

File: applications/rag/frontend/main.tf (hunk 6 above)
Comment (blackzlq) [reply applications/rag/frontend/main.tf:38]: We prefer here since the iap parts focus on k8s resource

File: modules/iap/iap.tf
Comment (blackzlq) [reply modules/iap/iap.tf:1]: Hmm, the system thought there are too many diff. I used move option

File: applications/jupyter/workloads-existing-brand-existing-client.example.tfvars [hunk 8]
```diff
@@ -26,22 +26,23 @@ cluster_membership_id = "" # required only for private cluster, default: cluster
...
+k8s_backend_service_port  = 80
 
-url_domain_addr   = ""
-url_domain_name   = ""
-client_id         = "<client_id>"
-client_secret     = "<client_secret>"
-members_allowlist = "allAuthenticatedUsers,user:<email>"
+url_domain_addr           = ""
+url_domain_name           = ""
+client_id                 = ""
```
Comment (umeshkumhar) [inline applications/jupyter/workloads-existing-brand-existing-client.example.tfvars:46]: revert this change

Comment (blackzlq) [reply applications/jupyter/workloads-existing-brand-existing-client.example.tfvars:46]: Good catch!

File: modules/iap/charts/iap/templates/static-ingress.yaml [hunk 9]
```diff
@@ -15,13 +15,14 @@
...
 kind: Ingress
 metadata:
-  name: {{ .Values.iap.jupyterIngress.name }}
+  name: {{ .Values.iap.ingress.name }}
   annotations:
-    kubernetes.io/ingress.global-static-ip-name:  {{ .Values.iap.jupyterIngress.staticIpName }}
-    networking.gke.io/managed-certificates:  {{ .Values.iap.jupyterIngress.managedCertificates }}
+    kubernetes.io/ingress.global-static-ip-name:  {{ .Values.iap.ingress.staticIpName }}
+    networking.gke.io/managed-certificates:  {{ .Values.iap.managedCertificate.name }}
+    kubernetes.io/ingress.class: "gce"
```
Comment (umeshkumhar) [inline modules/iap/charts/iap/templates/static-ingress.yaml:22]: do we need to mention this annotation? we already mention on service as neg and ingress true..

File: modules/iap/iap.tf [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+data "kubernetes_all_namespaces" "allns" {}
+
+resource "random_string" "random" {
+  length  = 4
+  special = false
+  upper   = false
+}
+
+# Jupyter IAP
+resource "google_compute_global_address" "jupyter_ip_address" {
```
Comment (umeshkumhar) [inline modules/iap/iap.tf:25]: keep generic resource name..

File: modules/iap/iap.tf [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+  provider     = google-beta
+  project      = var.project_id
+  name         = "jupyter-address-${random_string.random.result}"
+  address_type = "EXTERNAL"
+  ip_version   = "IPV4"
+}
+
+# Helm Chart IAP
+resource "helm_release" "iap_jupyter" {
+  count            = var.jupyter_add_auth ? 1 : 0
```
Comment (umeshkumhar) [inline modules/iap/iap.tf:36]: rename var jupyter_add_auth

File: modules/iap/iap.tf [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+  special = false
+  upper   = false
+}
+
+# Jupyter IAP
+resource "google_compute_global_address" "jupyter_ip_address" {
+  count        = var.jupyter_add_auth && var.jupyter_url_domain_addr == "" ? 1 : 0
+  provider     = google-beta
+  project      = var.project_id
+  name         = "jupyter-address-${random_string.random.result}"
```
Comment (umeshkumhar) [inline modules/iap/iap.tf:29]: keep name generic

File: modules/iap/iap.tf [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+  }
+
+  set {
+    name  = "iap.secret.client_secret"
+    value = base64encode(var.jupyter_client_secret)
+  }
+
+  set {
+    name = "iap.managedCertificate.name"
+    value = var.jupyter_k8s_managed_cert_name
```
Comment (umeshkumhar) [inline modules/iap/iap.tf:65]: remove all jupyter occurences from iap module..  jupyter keyword will appear in jupyter module only when calling iap module.

File: modules/iap/iap.tf [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+    value = var.jupyter_k8s_backend_service_name
+  }
+
+  set {
+    name  = "iap.ingress.backendServicePort"
+    value = var.jupyter_k8s_backend_service_port
+  }
+}
+
+resource "google_project_iam_member" "jupyter_project" {
```
Comment (umeshkumhar) [inline modules/iap/iap.tf:94]: same here

File: modules/iap/charts/iap/templates/static-ingress.yaml (hunk 9 above)
Comment (blackzlq) [reply modules/iap/charts/iap/templates/static-ingress.yaml:22]: Yes we do, the ingress true is helping to create the gcp resource like NEG. We still need to specify the k8s resource

File: modules/iap/iap.tf (hunk 11 above)
Comment (blackzlq) [reply modules/iap/iap.tf:36]: the same

File: modules/iap/iap.tf (hunk 13 above)
Comment (blackzlq) [reply modules/iap/iap.tf:65]: Discussed offline, will refactor it in next PR

File: applications/jupyter/variables.tf [hunk 15]
```diff
@@ -74,6 +85,12 @@ variable "k8s_backend_service_name" {
   default     = "proxy-public"
 }
 
+variable "k8s_backend_service_port" {
+  type        = number
+  description = "Name of the Backend Service on GCP"
+  default = 80
```
Comment (chiayi) [inline applications/jupyter/variables.tf:91]: NIT: Spacing

File: applications/jupyter/workloads-auto-create-brand.example.tfvars [hunk 16]
```diff
@@ -26,21 +26,23 @@ cluster_membership_id = "" # required only for private cluster, default: cluster
...
+workload_identity_service_account           = "jupyter-service-account"
 
 # Jupyterhub with IAP
-add_auth                = true
-brand                   = ""
-support_email           = "<email>"
-default_backend_service = "proxy-public"
-service_name            = "iap-config-default"
+add_auth                  = true
+brand                     = "projects/<prj-number>/brands/<prj-number>" # ensure brand is Internal
```
Comment (chiayi) [inline applications/jupyter/workloads-auto-create-brand.example.tfvars:36]: If this is the "auto-create-brand" file example, do we still need this template? If we're just going to have the same values/vars between the tfvar files I don't see the value added.

File: applications/jupyter/workloads-without-iap.example.tfvars [hunk 17]
```diff
@@ -32,15 +32,17 @@ create_service_account        = true
...
-k8s_backend_service_name = "proxy-public"
+k8s_backend_service_name  = "proxy-public"
+k8s_backend_service_port  = 80
 
-url_domain_addr = ""
-url_domain_name = ""
-client_id = ""
-client_secret = ""
-members_allowlist = ["allAuthenticatedUsers"]
+url_domain_addr           = ""
```
Comment (chiayi) [inline applications/jupyter/workloads-without-iap.example.tfvars:44]: Is it better to remove these iap specific lines completely for this example?

File: applications/rag/frontend/variables.tf [hunk 18]
```diff
@@ -66,4 +66,86 @@ variable "google_service_account" {
...
+variable "k8s_backend_service_name" {
+  type        = string
+  description = "Name of the K8s Backend Service, this is defined by Frontend"
+  default     = "rag-frontend"
+}
+
+variable "k8s_backend_service_port" {
+  type        = number
+  description = "Name of the K8s Backend Service Port"
+  default = 8080
```
Comment (chiayi) [inline applications/rag/frontend/variables.tf:108]: NIT: spacing

File: applications/rag/frontend/variables.tf [hunk 19]
```diff
@@ -66,4 +66,86 @@ variable "google_service_account" {
...
+variable "url_domain_name" {
+  type        = string
+  description = "Name of the domain provided by the user. This var will only be used if url_domain_addr is not empty"
+  default     = ""
+}
+
+variable "support_email" {
+  type        = string
+  description = "Email for users to contact with questions about their consent"
+  default = "<email>"
```
Comment (chiayi) [inline applications/rag/frontend/variables.tf:132]: NIT: same spacing here, probably in the other variable files too, assuming that you did not type out the the variable files.

**Review by chiayi — COMMENTED** [review]
Minor comments, will reach out offline on my question on the details of having to create two iap modules.

File: applications/jupyter/workloads-auto-create-brand.example.tfvars (hunk 16 above)
Comment (blackzlq) [reply applications/jupyter/workloads-auto-create-brand.example.tfvars]: Oops, good catch.

File: applications/jupyter/workloads-existing-brand-existing-client.example.tfvars [hunk 20]
```diff
@@ -26,22 +26,23 @@ cluster_membership_id = "" # required only for private cluster, default: cluster
...
 k8s_backend_config_name   = "jupyter-iap-config"
 k8s_backend_service_name  = "proxy-public"
+k8s_backend_service_port  = 80
 
 url_domain_addr   = ""
 url_domain_name   = ""
 client_id         = "<client_id>"
 client_secret     = "<client_secret>"
-members_allowlist = "allAuthenticatedUsers,user:<email>"
+members_allowlist = "allAuthenticatedUsers,user:<email>"
```
Comment (chiayi) [inline applications/jupyter/workloads-existing-brand-existing-client.example.tfvars:48]: Missing [ and ,

File: applications/rag/workloads.tfvars [hunk 21]
```diff
@@ -43,15 +42,36 @@ jupyter_service_account        = "jupyter-system-account"
...
+jupyter_k8s_iap_secret_name       = "jupyter-iap-secret"
+jupyter_k8s_backend_config_name   = "jupyter-iap-config"
+jupyter_k8s_backend_service_name  = "proxy-public"
+jupyter_k8s_backend_service_port  = 80
+
+jupyter_url_domain_addr   = ""
+jupyter_url_domain_name   = ""
+jupyter_client_id         = ""
+jupyter_client_secret     = ""
+jupyter_members_allowlist = ["user:zlq@google.com"]
```
Comment (chiayi) [inline applications/rag/workloads.tfvars:61]: Clean up. Also do we want "allAuthenticatedUsers" here as well?

File: applications/rag/workloads.tfvars [hunk 22]
```diff
@@ -43,15 +42,36 @@ jupyter_service_account        = "jupyter-system-account"
...
+frontend_k8s_iap_secret_name       = "frontend-iap-secret"
+frontend_k8s_backend_config_name   = "frontend-iap-config"
+frontend_k8s_backend_service_name  = "rag-frontend"
+frontend_k8s_backend_service_port  = 8080
+
+frontend_url_domain_addr   = ""
+frontend_url_domain_name   = ""
+frontend_client_id         = ""
+frontend_client_secret     = ""
+frontend_members_allowlist = ["user:<email>"]
```
Comment (chiayi) [inline applications/rag/workloads.tfvars:77]: Same here with "allAuthenticatedUsers"

**Review by chiayi — APPROVED** [review]
Few more minor comments, other than that LGTM

File: modules/iap/iap.tf (hunk 11 above)
Comment (bjornsen) [reply modules/iap/iap.tf:37]: Is it possible to implement the specifics for Jupyter outside of the more generic IAP module?

File: modules/jupyter/variables.tf [hunk 23]
```diff
@@ -47,7 +47,7 @@ variable "gcp_service_account_iam_roles" {
 variable "predefined_iam_roles" {
   description = "Predefined list of IAM roles to assign"
   type        = list(string)
-  default     = ["roles/compute.networkViewer", "roles/viewer", "roles/cloudsql.client", "roles/artifactregistry.reader", "roles/storage.admin", "roles/iam.serviceAccountAdmin", "roles/compute.loadBalancerServiceUser"]
+  default     = ["roles/compute.networkViewer", "roles/viewer", "roles/cloudsql.client", "roles/artifactregistry.reader", "roles/storage.admin", "roles/iam.serviceAccountAdmin", "roles/compute.loadBalancerServiceUser", "roles/iam.serviceAccountTokenCreator"]
```
Comment (bjornsen) [inline modules/jupyter/variables.tf:50]: This is becoming a very powerful account with serviceAccountTokenCreator, serviceAccountAdmin, and storage.admin. Can we cut down these permissions? At a minimum, we need to document why each of these permissions are required as defaults.

File: applications/jupyter/variables.tf (hunk 15 above)
[the remark this answers is not in the document]
Comment (blackzlq) [reply applications/jupyter/variables.tf]: this is the in cluster port, the load balancer is already https

File: applications/jupyter/workloads-auto-create-brand.example.tfvars (hunk 16 above)
[the remark this answers is not in the document]
Comment (blackzlq) [reply applications/jupyter/workloads-auto-create-brand.example.tfvars]: Good point, we can change it to a flag, make a todo here

File: applications/jupyter/workloads-auto-create-brand.example.tfvars [hunk 24]
```diff
@@ -26,21 +26,23 @@ cluster_membership_id = "" # required only for private cluster, default: cluster
...
-url_domain_addr   = ""
-url_domain_name   = ""
-client_id         = ""
-client_secret     = ""
-members_allowlist = "allAuthenticatedUsers,user:<email>"
+url_domain_addr           = ""
+url_domain_name           = ""
+client_id                 = ""
+client_secret             = ""
+members_allowlist         = ["allAuthenticatedUsers", "user:<email>"]
```
[the remark this answers is not in the document]
Comment (blackzlq) [reply applications/jupyter/workloads-auto-create-brand.example.tfvars:48]: https://cloud.google.com/iap/docs/managing-access#public_access

File: applications/jupyter/workloads-existing-brand-auto-create-client.example.tfvars [hunk 25]
```diff
@@ -26,21 +26,22 @@ cluster_membership_id = "" # required only for private cluster, default: cluster
...
-brand = "projects/<prj-number>/brands/<prj-number>" # ensure brand is Internal
-support_email = "<email>"
+add_auth                  = true
+brand                     = "projects/<prj-number>/brands/<prj-number>" # ensure brand is Internal
+support_email             = "<email>"
 k8s_ingress_name          = "jupyter-ingress"
+k8s_iap_secret_name       = "jupyter-iap-secret"
 k8s_backend_config_name   = "jupyter-iap-config"
 k8s_backend_service_name  = "proxy-public"
+k8s_backend_service_port  = 80
```
[the remark this answers is not in the document]
Comment (blackzlq) [reply applications/jupyter/workloads-existing-brand-auto-create-client.example.tfvars:41]: The same, in cluster port, not the public http, we have already onboard to iap https

File: applications/jupyter/workloads-existing-brand-auto-create-client.example.tfvars [hunk 26]
```diff
@@ -26,21 +26,22 @@ cluster_membership_id = "" # required only for private cluster, default: cluster
...
+namespace                                   = "jupyter"
+gcs_bucket                                  = "<gcs-bucket>"
+workload_identity_service_account           = "jupyter-service-account"
 
 # Jupyterhub with IAP
-add_auth = true
-brand = "projects/<prj-number>/brands/<prj-number>" # ensure brand is Internal
-support_email = "<email>"
+add_auth                  = true
+brand                     = "projects/<prj-number>/brands/<prj-number>" # ensure brand is Internal
```
[the remark this answers is not in the document]
Comment (blackzlq) [reply applications/jupyter/workloads-existing-brand-auto-create-client.example.tfvars:35]: the same

File: applications/rag/frontend/main.tf [hunk 27]
```diff
@@ -30,6 +34,46 @@ locals {
...
+
+  disable_dependent_services = false
+  disable_on_destroy         = false
+}
+
+# IAP Section: Creates the OAuth client used in IAP
+resource "google_iap_client" "iap_oauth_client" {
+  count        = var.client_id == "" ? 1 : 0
+  display_name = "Frontend-Client"
+  brand        = var.brand == "" ? "projects/${data.google_project.project.number}/brands/${data.google_project.project.number}" : var.brand
```
[the remark this answers is not in the document]
Comment (blackzlq) [reply applications/rag/frontend/main.tf:51]: Yes, clean this up in todo

File: applications/rag/frontend/main.tf (hunk 7 above)
[the remark this answers is not in the document]
Comment (blackzlq) [reply applications/rag/frontend/main.tf]: Yes, IAP required

File: modules/jupyter/variables.tf (hunk 23 above)
Comment (blackzlq) [reply modules/jupyter/variables.tf:50]: Add a todo to review all the permissions

File: applications/jupyter/workloads-existing-brand-existing-client.example.tfvars
[the remark this answers is not in the document]
Comment (blackzlq) [reply applications/jupyter/workloads-existing-brand-existing-client.example.tfvars:1]: We can discuss more, create a bug to track it. Lets focus on the IAP change first, this change is already too big. I don't want to introduce more change here. https://buganizer.corp.google.com/issues/327688037

File: applications/rag/frontend/variables.tf [hunk 28]
```diff
@@ -66,4 +66,86 @@ variable "google_service_account" {
...
+variable "k8s_backend_service_port" {
+  type        = number
+  description = "Name of the K8s Backend Service Port"
+  default     = 8080
+}
+
+variable "brand" {
+  type        = string
+  description = "name of the brand if there isn't already on the project. If there is already a brand for your project, please leave it blank and empty"
+  default     = ""
```
Comment (umeshkumhar) [inline applications/rag/frontend/variables.tf:114]: FYI, if the brand var is empty.. we also attempt to create a brand..

File: applications/rag/frontend/outputs.tf [hunk 29]
```diff
@@ -12,6 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-output "domain" {
-  value = var.url_domain_addr != "" ? var.url_domain_addr : "${google_compute_global_address.default[0].address}.nip.io"
-}
+output "frontend_domain" {
+  value = var.add_auth ? module.iap_auth[0].frontend_domain : var.url_domain_addr
```
Comment (umeshkumhar) [inline applications/rag/frontend/outputs.tf:16]: `var.url_domain_addr` will not make sense here.. because this variable is only set for IAP purpose.

if IAP is false.. this var will be false always.. instead lets fetch ip from frontend-service (if its loadbalancer type..) or else just a text that "internal_access_only"

Comment (blackzlq) [reply applications/rag/frontend/outputs.tf:16]: Good point, let me make it as jupyter

File: applications/rag/frontend/variables.tf (hunk 28 above)
Comment (blackzlq) [reply applications/rag/frontend/variables.tf:114]: Yeah, the brand will be set in rag

File: applications/rag/workloads.tfvars [hunk 30]
```diff
@@ -44,15 +43,36 @@ jupyter_service_account        = "jupyter-system-account"
...
-k8s_backend_service_name  = "proxy-public"
-
-url_domain_addr   = ""
-url_domain_name   = ""
-client_id         = ""
-client_secret     = ""
-members_allowlist = ["allAuthenticatedUsers", "user:<email>"]
+
+## Jupyter IAP Settings
+jupyter_add_auth                  = true # Set to true when using auth with IAP
```
Comment (umeshkumhar) [inline applications/rag/workloads.tfvars:49]: when we say enable IAP on rag.. that bydefault means that jupyter and rag frontend .. both needs same IAP configs.. like add_auth, support_email,

File: modules/iap/iap.tf (hunk 10 above)
Comment (umeshkumhar) [reply modules/iap/iap.tf:27]: yes

File: modules/iap/iap.tf [hunk 31]
```diff
@@ -0,0 +1,178 @@
...
+  for_each = toset(var.jupyter_members_allowlist)
+  project  = var.project_id
+  role     = "roles/iap.httpsResourceAccessor"
+  member   = each.key
```
Comment (umeshkumhar) [inline modules/iap/iap.tf:100]: I think this doesnot support - allAuthenticatedUsers..

may be for now we can remove non-supported. or may be replace this block with old service_backend_iam resource?

File: applications/rag/workloads.tfvars (hunk 30 above)
Comment (blackzlq) [reply applications/rag/workloads.tfvars:49]: I think add auth and support email can be different. Customer can only enable the jupyter or frontend, and the support team for those two can be different

File: modules/iap/iap.tf (hunk 31 above)
Comment (blackzlq) [reply modules/iap/iap.tf:100]: Sure I will remove it and leave a todo

File: applications/rag/workloads.tfvars [hunk 32]
```diff
@@ -20,9 +20,8 @@ cluster_name     = "<cluster_name>"
 cluster_location = "us-central1"
 
 ## GKE environment variables
-kubernetes_namespace = "rag"
-create_gcs_bucket    = true
```
Comment (umeshkumhar) [inline applications/rag/workloads.tfvars:24]: this is required.

**Review by umeshkumhar — DISMISSED** [review]
lgtm

**Review by umeshkumhar — APPROVED** [review]
lgtm

**Review by imreddy13 — APPROVED** [review]
